From 5183b36e754cf1c5314a0d9561272d330bf5f6e5 Mon Sep 17 00:00:00 2001
From: Vincent Cheng <vcheng@debian.org>
Date: Sat, 19 Apr 2025 13:08:34 +0200
Subject: [PATCH 1/4] Fix search path for pyrogenesis

Modify launcher script to execute /usr/games/pyrogenesis if /usr/games is not in the user's $PATH.

Forwarded: http://trac.wildfiregames.com/ticket/1424
Author: Vincent Cheng <vcheng@debian.org>
Bug-Debian: https://bugs.debian.org/679033
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1380737
Last-Update: 2025-02-01
---
 build/resources/0ad.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build/resources/0ad.sh b/build/resources/0ad.sh
index 5d2f0dde91..1941be2471 100755
--- a/build/resources/0ad.sh
+++ b/build/resources/0ad.sh
@@ -3,6 +3,9 @@
 pyrogenesis=$(which pyrogenesis 2>/dev/null)
 if [ -x "$pyrogenesis" ]; then
 	"$pyrogenesis" "$@"
+elif [ -x "/usr/games/pyrogenesis" ]; then
+	# Fallback in case /usr/games is not in $PATH; see #679033 and LP: #1380737
+	/usr/games/pyrogenesis "$@"
 else
 	echo "Error: pyrogenesis not found in ($PATH)"
 	exit 1
-- 
2.49.0

